tests/test_benchmark.py

```diff
@@ -0,0 +1,212 @@
+# limitations under the License.
+import os
+from unittest import TestCase
+from datetime import timedelta
+import numpy as np
+import pandas as pd
+from testfixtures import TempDirectory
+from zipline.data.us_equity_pricing import SQLiteAdjustmentWriter, \
+    SQLiteAdjustmentReader
+from zipline.errors import (
+    BenchmarkAssetNotAvailableTooEarly,
+    BenchmarkAssetNotAvailableTooLate,
+    InvalidBenchmarkAsset)
+
+from zipline.finance.trading import TradingEnvironment
+from zipline.gens.tradesimulation import AlgorithmSimulator
+from zipline.utils import factory
+from zipline.utils.test_utils import create_data_portal, write_minute_data
+from .test_perf_tracking import MockDailyBarSpotReader
+
+
+class TestBenchmark(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.env = TradingEnvironment()
+        cls.tempdir = TempDirectory()
+
+        cls.sim_params = factory.create_simulation_parameters()
+
+        cls.env.write_data(equities_data={
+            1: {
+                "start_date": cls.sim_params.trading_days[0],
+                "end_date": cls.sim_params.trading_days[-1] + timedelta(days=1)
+            },
+            2: {
+                "start_date": cls.sim_params.trading_days[0],
+                "end_date": cls.sim_params.trading_days[-1] + timedelta(days=1)
+            },
+            3: {
+                "start_date": cls.sim_params.trading_days[100],
+                "end_date": cls.sim_params.trading_days[-100]
+            },
+            4: {
+                "start_date": cls.sim_params.trading_days[0],
+                "end_date": cls.sim_params.trading_days[-1] + timedelta(days=1)
+            }
+
+        })
+
+        dbpath = os.path.join(cls.tempdir.path, "adjustments.db")
+
+        writer = SQLiteAdjustmentWriter(dbpath, cls.env.trading_days,
+                                        MockDailyBarSpotReader())
+        splits = mergers = pd.DataFrame(
+            {
+                # Hackery to make the dtypes correct on an empty frame.
+                'effective_date': np.array([], dtype=int),
+                'ratio': np.array([], dtype=float),
+                'sid': np.array([], dtype=int),
+            },
+            index=pd.DatetimeIndex([], tz='UTC'),
+            columns=['effective_date', 'ratio', 'sid'],
+        )
+        dividends = pd.DataFrame({
+            'sid': np.array([], dtype=np.uint32),
+            'amount': np.array([], dtype=np.float64),
+            'declared_date': np.array([], dtype='datetime64[ns]'),
+            'ex_date': np.array([], dtype='datetime64[ns]'),
+            'pay_date': np.array([], dtype='datetime64[ns]'),
+            'record_date': np.array([], dtype='datetime64[ns]'),
+        })
+        declared_date = cls.sim_params.trading_days[45]
+        ex_date = cls.sim_params.trading_days[50]
+        record_date = pay_date = cls.sim_params.trading_days[55]
+
+        stock_dividends = pd.DataFrame({
```

Comment:
The point of this part is to support testing that we don't support equities with a stock dividend as benchmark (see last test in this suite).